docs-devsite/firestore_.firestoredataconverter.md

```diff
@@ -98,43 +98,162 @@ toFirestore(modelObject: PartialWithFieldValue<AppModelType>, options: SetOption
+}
 
-const postSnap = await firebase.firestore()
-  .collection('posts')
-  .withConverter(postConverter)
-  .doc().get();
-const post = postSnap.data();
-if (post !== undefined) {
-  post.title; // string
-  post.toString(); // Should be defined
-  post.someNonExistentProperty; // TS error
+async function advancedDemo(db: Firestore): Promise<void> {
+    // Create a `DocumentReference` with a `FirestoreDataConverter`.
+    const documentRef = doc(db, 'posts/post123').withConverter(new PostConverter());
+
+    // The `data` argument specified to `setDoc()` is type checked by the
+    // TypeScript compiler to be compatible with `Post`. Since the `data`
+    // argument is typed as `WithFieldValue<Post>` rather than just `Post`,
+    // this allows properties of the `data` argument to also be special
+    // Firestore values that perform server-side mutations, such as
+    // `arrayRemove()`, `deleteField()`, and `serverTimestamp()`.
+    await setDoc(documentRef, {
+        title: 'My Life',
+        author: 'Foo Bar',
+        lastUpdatedMillis: serverTimestamp()
+    });
+
+    // The TypeScript compiler will fail to compile if the `data` argument to
+    // `setDoc()` is _not_ be compatible with `WithFieldValue<Post>`. This
+    // type checking prevents the caller from specifying objects with incorrect
+    // properties or property values.
+    // @ts-expect-error "Argument of type { ttl: string; } is not assignable
+    // to parameter of type WithFieldValue<Post>"
+    await setDoc(documentRef, { ttl: 'The Title' });
+
+    // When retrieving a document with `getDoc()` the `DocumentSnapshot`
+    // object's `data()` method returns a `Post`, rather than a generic object,
+    // which would have been returned if the `DocumentReference` did _not_ have a
+    // `FirestoreDataConverter` attached to it.
+    const snapshot1: DocumentSnapshot<Post> = await getDoc(documentRef);
+    const post1: Post = snapshot1.data()!;
+    if (post1) {
+        assertEqual(post1.title, 'My Life');
+        assertEqual(post1.author, 'Foo Bar');
+    }
+
+    // The `data` argument specified to `updateDoc()` is type checked by the
+    // TypeScript compiler to be compatible with `PostDbModel`. Note that
+    // unlike `setDoc()`, whose `data` argument must be compatible with `Post`,
+    // the `data` argument to `updateDoc()` must be compatible with
+    // `PostDbModel`. Similar to `setDoc()`, since the `data` argument is typed
+    // as `WithFieldValue<PostDbModel>` rather than just `PostDbModel`, this
+    // allows properties of the `data` argument to also be those special
+    // Firestore values, like as `arrayRemove()`, `deleteField()`, and
+    // `serverTimestamp()`.
+    await updateDoc(documentRef, {
+        'aut.firstName': 'NewFirstName',
+        lut: serverTimestamp()
+    });
+
+    // The TypeScript compiler will fail to compile if the `data` argument to
+    // `updateDoc()` is _not_ be compatible with `WithFieldValue<PostDbModel>`.
```

Comment:
Remove "be": "...is not compatible with..."

Reply:
TY